tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/passes/ConfigurationPassImpl.scala

```diff
@@ -96,14 +130,46 @@ class ConfigurationPassImpl @Inject()(val options: PassOptions,
         throw new TLCConfigurationError("  > Could not parse TLC configuration in " + basename + ": " + e.msg)
     }
 
-    val rewritten = new ConstAndDefRewriter(tracker)(module)
+    // rewrite constants and declarations
+    new ConstAndDefRewriter(tracker)(configuredModule)
+  }
 
-    // dump the result of preprocessing
-    val outdir = options.getOrError("io", "outdir").asInstanceOf[Path]
-    PrettyWriter.write(rewritten, new File(outdir.toFile, "out-config.tla"))
+  // Make sure that all operators passed via --init, --cinit, --next, --inv are present.
+  private def ensureDeclarationsArePresent(mod: TlaModule): Unit = {
+    def assertDecl(role: String, name: String): Unit = {
```

Comment:
We might dry this out a bit, and, imo, make the intention clearer to readers, by moving the "optionality" logic into this helper. E.g., make `assertDecl` something like

```suggestion
    def assertDecl(role: String, name: Option[String], optiona: Bool): Unit = {
```

And then move all the checking logic here, then making the checks just look like

```scala
assertDecl("the initialization predicate", options.get[String]("checker", "init"), optional = False)
assertDecl("the constant initialization predicate", options.get[String]("checker", "cinit"), optional = True)
...
```

Reply:
I see what you mean about the invariants case, that might undercut any gains in readability. I agree its clear enough.